sdk/storage/storage-blob/src/BlobBatchClient.ts

```diff
@@ -0,0 +1,279 @@
+   * @returns {Promise<BlobBatchSetBlobsAccessTierResponse>}
+   * @memberof BatchSetTierRequest
+   */
+  public async setBlobsAccessTier(
+    urls: string[],
+    credential: SharedKeyCredential | AnonymousCredential | TokenCredential,
+    tier: Models.AccessTier,
+    options?: BlobSetTierOptions
+  ): Promise<BlobBatchSetBlobsAccessTierResponse>;
+
+  /**
+   * Create multiple set tier operations to set the tier on a blob.
+   * The operation is allowed on a page blob in a premium
+   * storage account and on a block blob in a blob storage account (locally redundant
+   * storage only). A premium page blob's tier determines the allowed size, IOPS,
+   * and bandwidth of the blob. A block blob's tier determines Hot/Cool/Archive
+   * storage type. This operation does not update the blob's ETag.
+   * See [set blob tier details](https://docs.microsoft.com/en-us/rest/api/storageservices/set-blob-tier).
+   * The operation(subrequest) will be authenticated and authorized
+   * with specified credential.See [blob batch authorization details](https://docs.microsoft.com/en-us/rest/api/storageservices/blob-batch#authorization).
+   *
+   * @param {BlobClient[]} blobClients The BlobClients for the blobs which should have a new tier set.
+   * @param {Models.AccessTier} tier
+   * @param {BlobSetTierOptions} [options]
+   * @returns {Promise<BlobBatchSetBlobsAccessTierResponse>}
+   * @memberof BatchSetTierRequest
+   */
+  public async setBlobsAccessTier(
+    blobClients: BlobClient[],
+    tier: Models.AccessTier,
+    options?: BlobSetTierOptions
+  ): Promise<BlobBatchSetBlobsAccessTierResponse>;
+
+  public async setBlobsAccessTier(
+    urlsOrBlobClients: string[] | BlobClient[],
+    credentialOrTier:
+      | SharedKeyCredential
+      | AnonymousCredential
+      | TokenCredential
+      | Models.AccessTier,
+    tierOrOptions?: Models.AccessTier | BlobSetTierOptions,
+    options?: BlobSetTierOptions
+  ): Promise<BlobBatchSetBlobsAccessTierResponse> {
+    const batch = new BlobBatch();
+    for (const urlOrBlobClient of urlsOrBlobClients) {
+      if (typeof urlOrBlobClient === "string") {
+        await batch.setBlobAccessTier(
+          urlOrBlobClient,
+          credentialOrTier as TokenCredential,
+          tierOrOptions as Models.AccessTier,
+          options
+        );
+      } else {
+        await batch.setBlobAccessTier(
+          urlOrBlobClient,
+          credentialOrTier as Models.AccessTier,
+          tierOrOptions as BlobSetTierOptions
+        );
+      }
+    }
+    return this.submitBatch(batch);
+  }
+
+  /**
+   * Submit batch request which consists of multiple subrequests.
+   *
+   * @example
+   * ```js
+   * let batchRequest = new BlobBatch();
+   * await batchRequest.deleteBlob(urlInString, credential, {
+   *  deleteSnapshots: "include"
+   * });
+   * await batchRequest.setBlobAccessTier(blockBlobClient, "Cool", {
+   *  leaseAccessConditions: { leaseId: leaseId }
```

Comment:
Have you tested if this works? As I know batch request can I only support one and only one kind of operation.

Reply:
Right the service currently does not allow it.  .NET and Java throws exception when different types of requests are added. We could do something similar?

Reply:
Here's what .NET does https://github.com/Azure/azure-sdk-for-net/blob/7c2ff45b0f4c60fc9ed2420969b556851e74cd4d/sdk/storage/Azure.Storage.Blobs.Batching/src/BlobBatch.cs#L87

We should add similar checks and if service support comes in the future remove the checks.